libparsec/crates/platform_device_loader/src/native/mod.rs

```diff
@@ -322,17 +327,50 @@ fn save_device_file(key_file: &PathBuf, file_content: &DeviceFile) -> Result<(),
     Ok(())
 }
 
+fn generate_keyring_user(
+    keyring_user_path: &PathBuf,
+) -> Result<(SecretKey, String), SaveDeviceError> {
+    // Generate a keyring user
+    let keyring_user = Uuid::new_v4().to_string();
+
+    // Generate a key
+    let (passphrase, key) = SecretKey::generate_recovery_passphrase();
+
+    let entry = KeyringEntry::new(KEYRING_SERVICE, &keyring_user)?;
+
+    // Add the key to the keyring
+    entry.set_password(&passphrase)?;
+
+    // Save the keyring user to the config file
+    save_content(keyring_user_path, keyring_user.as_bytes())?;
+
+    Ok((key, keyring_user))
+}
+
 pub async fn save_device(
     access: &DeviceAccessStrategy,
     device: &LocalDevice,
 ) -> Result<(), SaveDeviceError> {
     match access {
         DeviceAccessStrategy::Keyring { key_file } => {
-            let entry = KeyringEntry::new(KEYRING_SERVICE, &device.slughash())?;
-
-            let (passphrase, key) = SecretKey::generate_recovery_passphrase();
-
-            entry.set_password(&passphrase)?;
+            let keyring_user_path = get_default_config_dir().join("keyring_user.txt");
```

Comment:
I'm thinking if it would be better to store the file under `data_dir` :thinking:

I don't think that something that will be configured.

Reply:
> I'm thinking if it would be better to store the file under data_dir 🤔

I would keep it close to the device files. In v2, device files are stored in the config directory, but I don't know about v3. What do you think @FirelightFlagboy ?

Reply:
Storing the device in the data dir would be nice :nerd_face: